$OpenBSD: patch-cups_http_c,v 1.8 2011/10/15 07:20:43 ajacoutot Exp $

Increase timeout for the PPD file to be copied so that it doesn't fail
on low-end configurations.

Fix build on gcc-2.95 archs.

--- cups/http.c.orig	Thu Jun 16 22:12:16 2011
+++ cups/http.c	Sat Oct 15 09:18:38 2011
@@ -1314,7 +1314,7 @@ httpGets(char   *line,			/* I - Line to read into */
       * No newline; see if there is more data to be read...
       */
 
-      while (!_httpWait(http, http->blocking ? 30000 : 10000, 1))
+      while (!_httpWait(http, http->blocking ? 90000 : 10000, 1))
       {
 	if (http->timeout_cb && (*http->timeout_cb)(http, http->timeout_data))
 	  continue;
@@ -1469,6 +1469,8 @@ httpInitialize(void)
   static int	initialized = 0;	/* Have we been called before? */
 #ifdef WIN32
   WSADATA	winsockdata;		/* WinSock data */
+#elif defined(HAVE_SIGACTION)
+  struct sigaction	action;		/* POSIX sigaction data */
 #endif /* WIN32 */
 #ifdef HAVE_LIBSSL
   int		i;			/* Looping var */
@@ -1495,7 +1497,6 @@ httpInitialize(void)
   sigset(SIGPIPE, SIG_IGN);
 
 #  elif defined(HAVE_SIGACTION)
-  struct sigaction	action;		/* POSIX sigaction data */
 
 
   memset(&action, 0, sizeof(action));
